src/test/codegen/codemodels.rs

```diff
@@ -0,0 +1,18 @@
+// revisions: NOMODEL MODEL-SMALL MODEL-KERNEL MODEL-MEDIUM MODEL-LARGE
+//[NOMODEL] compile-flags:
+//[MODEL-SMALL] compile-flags: -C code-model=small
+//[MODEL-KERNEL] compile-flags: --target x86_64-unknown-linux-gnu -C code-model=kernel
```

Comment:
Pretty sure this is going to fail because not all builder have an x86_64-unknown-linux-gnu target (in particular, those where this is not the host target).

You need to either make this test `only-x86_64`, or drop the kernel code model part of the test, if that's not portable to other targets.

Reply:
There's an alternative option of using:

```
// needs-llvm-components:
```

together with a `#[no_core]` example.